packages/calcite-components/src/utils/dom.ts

```diff
@@ -200,20 +200,27 @@ export function queryElementRoots<T extends Element = Element>(
 /**
  * This helper returns the closest element matching the selector by crossing he shadow boundary if necessary.
  *
+ * Based on https://stackoverflow.com/q/54520554/194216
+ *
  * @param {Element} element The starting element.
  * @param {string} selector The selector.
  * @returns {Element} The targeted element.
  */
+export function closestElementCrossShadowBoundary<TagName extends keyof HTMLElementTagNameMap>(
+  element: Element,
+  selector: TagName,
+): HTMLElementTagNameMap[TagName] | null;
+export function closestElementCrossShadowBoundary<T extends Element = Element>(
+  element: Element,
+  selector: string,
+): T | null;
 export function closestElementCrossShadowBoundary<T extends Element = Element>(
   element: Element,
   selector: string,
 ): T | null {
-  // based on https://stackoverflow.com/q/54520554/194216
-  function closestFrom<T extends Element = Element>(el: Element): T | null {
```

Comment:
It appears that creating this function was needless.

We can get a small performance boost by not creating functions on the fly (as then we can benefit from v8 caching). More info:

https://marvinh.dev/blog/speeding-up-javascript-ecosystem/#:~:text=Inline%20functions%2C%20inline%20caches%20and%20recursion

Reply:
Nice cleanup! The maintainability win is a great bonus too.